src/realm/table.cpp

```diff
@@ -3257,8 +3257,6 @@ void Table::set_binary(size_t col_ndx, size_t ndx, BinaryData value, bool is_def
         throw LogicError(LogicError::column_index_out_of_range);
     if (!is_nullable(col_ndx) && value.is_null())
         throw LogicError(LogicError::column_not_nullable);
-    if (REALM_UNLIKELY(value.size() > ArrayBlob::max_binary_size))
-        throw LogicError(LogicError::binary_too_big);
```

Comment:
The same size check happens in transact_log.hpp line 2246. This means that if sync uses the `set_binary_big` operation the client that first creates the big data will succeed, but all other clients will crash with a bad transaction log exception. Please discuss with @simonask or another sync team member if the check can simply be removed there as well, or if we need another instruction for `set_binary_big`. Adding a test for handling this new instruction through the transaction logs would also be a good idea.